test/unit/process/test_cuda_process_set.cpp

```diff
@@ -0,0 +1,128 @@
+{
+
+  auto get_n_react = std::bind(std::uniform_int_distribution<>(0, 3), std::default_random_engine());
+  auto get_n_product = std::bind(std::uniform_int_distribution<>(0, 10), std::default_random_engine());
+  auto get_species_id = std::bind(std::uniform_int_distribution<>(0, n_species - 1), std::default_random_engine());
+  std::lognormal_distribution<double> distribution(-2.0, 4.0);
+  auto get_double = std::bind(distribution, std::default_random_engine());
+
+  std::vector<micm::Species> species{};
+  std::vector<std::string> species_names{};
+  for (std::size_t i = 0; i < n_species; ++i)
+  {
+    species.push_back(micm::Species{ std::to_string(i) });
+    species_names.push_back(std::to_string(i));
+  }
+  micm::Phase gas_phase{ species };
+  micm::State<MatrixPolicy> state{ micm::StateParameters{ .state_variable_names_{ species_names },
+                                                          .number_of_grid_cells_ = n_cells,
+                                                          .number_of_custom_parameters_ = 0,
+                                                          .number_of_rate_constants_ = n_reactions } };
+
+  std::vector<micm::Process> processes{};
+  for (std::size_t i = 0; i < n_reactions; ++i)
+  {
+    auto n_react = get_n_react();
+    std::vector<micm::Species> reactants{};
+    for (std::size_t i_react = 0; i_react < n_react; ++i_react)
+    {
+      reactants.push_back({ std::to_string(get_species_id()) });
+    }
+    auto n_product = get_n_product();
+    std::vector<yields> products{};
+    for (std::size_t i_prod = 0; i_prod < n_product; ++i_prod)
+    {
+      products.push_back(yields(std::to_string(get_species_id()), 1.2));
+    }
+    processes.push_back(micm::Process::create().reactants(reactants).products(products).phase(gas_phase));
+  }
+
+  micm::ProcessSet set{ processes, state };
+
+  for (auto& elem : state.variables_.AsVector())
+    elem = get_double();
+
+  MatrixPolicy<double> rate_constants{ n_cells, n_reactions };
+  for (auto& elem : rate_constants.AsVector())
+    elem = get_double();
+
+  MatrixPolicy<double> cpu_forcing{ n_cells, n_species, 1000.0};
+  MatrixPolicy<double> gpu_forcing{ };
+  gpu_forcing = cpu_forcing; 
+
+  
+  //kernel function call 
+  // double t0 = 0.0; 
+  // for (int i = 0; i < 100; i++)
+  // {
+  //   auto start = std::chrono::steady_clock::now(); 
+  //   set.CudaAddForcingTerms(rate_constants, state.variables_, gpu_forcing); 
+  //   auto end = std::chrono::steady_clock::now(); 
+  //   std::chrono::duration<double> duration = end - start;
+  //   t0 = t0 + duration.count(); 
+  // }
+  // std::cout << "time performance: "<< t0/100 <<std::endl; 
+
+  //kernel function call 
+  set.CudaAddForcingTerms<MatrixPolicy>(rate_constants, state.variables_, gpu_forcing); 
+    
+  //CPU function call
+  set.AddForcingTerms<MatrixPolicy>(rate_constants, state.variables_, cpu_forcing); 
```

Comment:
I would suggest adding a similar loop for the CPU call so that we could measure the CPU performance easily, too.

Reply:
If it is desired to put the performance measurement in a separate branch/repo, just discard this comment.